akka-persistence/src/main/scala/akka/persistence/Eventsourced.scala

```diff
@@ -774,8 +774,15 @@ private[persistence] trait Eventsourced
             sequenceNr = highestSeqNr
             setLastSequenceNr(highestSeqNr)
             _recoveryRunning = false
-            try Eventsourced.super.aroundReceive(recoveryBehavior, RecoveryCompleted)
-            finally transitToProcessingState() // in finally in case exception and resume strategy
+            try {
+              Eventsourced.super.aroundReceive(recoveryBehavior, RecoveryCompleted)
+              transitToProcessingState()
+            } catch {
+              case NonFatal(t) =>
+                try onRecoveryFailure(t, Some(RecoveryCompleted))
+                finally context.stop(self)
+            }
```

Comment:
The original intended behavior was to handle exceptions from RecoveryCompleted with ordinary supervision. I don't think it's the right thing to change that to forcefully stop the actor. For journal (db) errors we have a strong reason for doing that but exception from RecoveryCompleted is an application level problem and Akka can't really decide how that is best handled.

If you don't want to add supervision in the parent for this case you could try-catch in your RecoveryCompleted processing and throw `ActorInitializationException`. That is by default handled by supervision to stop the actor.